nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -211,6 +213,25 @@ public NiFiProperties get() {
         return instance;
     }
 
+    private void checkForDuplicates(File file) {
+        if (file == null || !file.exists() || !file.canRead()) {
+            throw new IllegalArgumentException("NiFi properties file missing or unreadable");
+        }
+
+        Props properties = new Props();
+        try (final InputStream inputStream = new BufferedInputStream(new FileInputStream(file))) {
+            properties.load(inputStream);
+        } catch (final Exception e) {
+            throw new RuntimeException(String.format("Loading Application Properties [%s] failed", file), e);
+        }
+
+        Set<String> duplicateKeys = properties.duplicateKeySet();
+        if (!duplicateKeys.isEmpty()) {
+            duplicateKeys.forEach(v -> logger.error("Multiple entries with different values found for key '{}'.", v));
+            throw new IllegalArgumentException("Duplicate keys were detected in the properties file. See previous errors.");
```

Comment:
Now that error handling includes only property keys, recommend aggregating the error into the exception message, as opposed to multiple log statements:
```suggestion
            throw new IllegalArgumentException(String.format("Application Properties contains duplicate property names [%s]", String.join(",", duplicateKeys));
```

Reply:
Log messages have been aggregated.